core/src/main/java/hudson/lifecycle/UnixLifecycle.java

```diff
@@ -42,11 +43,14 @@
  *
  * <p>
  * Restart by exec to self.
+ * The restartCommand configuration may override executbale name.
  *
  * @author Kohsuke Kawaguchi
  * @since 1.304
  */
 public class UnixLifecycle extends Lifecycle {
+    public static String RESTART_COMMAND = Configuration.getStringConfigParameter("restartCommand", null);
```

Comment:
Maybe also `@Resticted(NoExternalUse.class)` if you do not want to make it a part of public API for plugins (no need for it AFAICT)

Reply:
So yes, the field should be either Restricted or final (the latter for read-only API)

Reply:
ok